### ISSUE TITLE
Mitsuba 3.6 transformation toolbox confusing error when using removed static functions

### ISSUE THREAD

**Issue by DanielPSchenk** [mitsuba-renderer/mitsuba3#1409]
## Summary

The guide on porting to Mitsuba 3.6 states that the static functions of mitsuba.Transform4f were removed. The functions seem to still be around, although not working anymore, and throw type errors that do not help with debugging.

## Description
Code like 
```
mi.ScalarTransform4f.scale([2, 2, 2])
```
throws
```
TypeError: scale(): incompatible function arguments. The following argument types are supported:
    1. scale(self, v: mitsuba.ScalarPoint3f) -> mitsuba.ScalarTransform4f

Invoked with types: list
```
When trying to "fix" this error like this
```
mi.ScalarTransform4f.scale(mi.ScalarPoint3f([2, 2, 2]))
```
an even more confusing error occurs:
```
TypeError: scale(): incompatible function arguments. The following argument types are supported:
    1. scale(self, v: mitsuba.ScalarPoint3f) -> mitsuba.ScalarTransform4f

Invoked with types: mitsuba.ScalarPoint3f
```
To people who haven't read the porting guide, I think this would lead to total confusion (and maybe some issues on GitHub). 

It seems like the plugin documentation was not updated, see for example the shape plugin documentation (where I got the example line from).

**Comment by DanielPSchenk** [comment]
Last paragraph is fixed now with 9f9fd8e. Thank you @rtabbara !

**Comment by rtabbara** [comment]
Hi @DanielPSchenk ,

Thanks for pointing out the issue with the documentation! It should all hopefully be fixed now.

The error maybe is somewhat cryptic but that's just a consequence of Python. E.g.

```
class Foo():
    def __init__(self):
        self.a = 3

    def add(self):
        self.a += 1


b = Foo()
# These are equivalent
b.add()
Foo.add(b)

# TypeError: Foo.add() missing 1 required positional argument: 'self'
Foo.add()
```

which I think is consistent with the `TypeError` we get in your example. i.e. the `self` argument is missing.
